Error Response Codes Mapping (#2128)
TykTechnologies/tyk-docs: tyk-docs/content/error-response-codes.md

---
title: "Error Response Codes"
date: 2023-1-6
tags: ["error codes", "error response codes"]
description: "Response codes that the Tyk Gateway (v4.3.0) returns via an API"
weight: 201
menu:
  main:
    parent: Tyk Gateway
---

> This table contains the error codes that could possibly show up when using the Gateway. Each error code has a text and a description of what could have caused the error. **As Tyk continues to grow, so will the error response codes. As such, we will do our best to coordinate them on this page.**

Reviewer: Please reword to a formal tone

Reviewer: maybe @chrisanderton  can advise

Reviewer 2: Should we call them "Error response _status_ codes" to be consistent with [HTTP language](https://httpwg.org/specs/rfc9110.html#overview.of.status.codes)?

We could probably borrow some of that for the intro too..?

Probably shouldn't say "we'll do our best" as it doesn't inspire confidence. In the olden days, this kind of table would be generated directly from code, as the dictionary of codes and descriptions would be in a single file with constants and used consistently.. do we do that here?

<hr/>

Responses from the Gateway include a three-digit status code that describes the result of the request and the semantics of the response. The first digit of the status code defines the class of response. The error status codes used by the Gateway are defined in the table below.

- **1xx (Informational):** The request was received, continuing process
- **2xx (Successful)**: The request was successfully received, understood, and accepted
- **3xx (Redirection)**: Further action needs to be taken in order to complete the request
- [**4xx (Client Error)**](): The request contains bad syntax or cannot be fulfilled
- [**5xx (Server Error)**](): The server failed to fulfill an apparently valid request

Author: To my knowledge, i dont think theres a table that can be generated directly from the code. These codes were manually found within the code so there could be still some error codes missing from this table.

Author: looks good to merge for me!

> | Code: | Text                                                                                    | Description                                                                                                                                                                                                                                                                                                                                                                                                      |

Reviewer 3: Just Code, not Code:

> | Code: | Text                                                                                    | Description                                                                                                                                                                                                                                                                                                                                                                                                      |

Reviewer: ```suggestion
| Code | Text                                                                                    | Description                                                                                                                                                                                                                                                                                                                                                                                                      |
```

| 504   | Upstream service reached hard timeout                                                   | Timeout awaiting response headers during a request round trip.                                                                                                                                                                                                                                                                                                                                                   |
| 507   | Status Insufficient Storage                                                             | Attempted to update an API through a POST request but failed to due insufficient storage.                                                                                                                                                                                                                                                                                                                        |

> | x509  | Certificate signed by unknown authority                                                 | A client accessing Tyk with full ControlAPI access without a certificate.                                                                                                                                                                                                                                                                                                                                        |

Reviewer 3: Not a response code so I would get rid of x509

Reviewer: Not sure I understand

> | x509  | Certificate signed by unknown authority                                                 | A client accessing Tyk with full ControlAPI access without a certificate.                                                                                                                                                                                                                                                                                                                                        |

Reviewer: ```suggestion                                                                                                                                                                                                                                                                                                                               |
```

---
title: "Error Response Status Codes"
date: 2023-1-6
tags: ["error codes", "error status response codes"]

> description: "Error response codes that the Tyk Gateway (v4.3.0) returns via an API"

Reviewer: ```suggestion
description: "Error response codes that the Tyk Gateway returns via an API"
```

weight: 201
menu:
  main:
    parent: Tyk Gateway
---

> Responses from the Gateway include a three-digit status code that describes the result of the request and the semantics of the response. The first digit of the status code defines the class of response. The error status codes used by the Gateway are defined in the table below.

Reviewer: ```suggestion
Tyk gateway responses include HTTP status codes that follow the [HTTP standard](https://datatracker.ietf.org/doc/html/rfc9110). They have three digits that describe the result of the request and the semantics of the response. The first digit defines the class of response as shown in the [list](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) below:
```
